#2389 - Bracket is missing when opening a file RXN v2000/v3000 files with 'S-Group Properties Type = Multiple group'

vitaepam opened the pull request:
## How the feature works? / How did you fix the issue?
- 'DAT' S-Groups did not implement function which draw brackets for such type of groups
- replaced packages/ketcher-core/src/application/render/restruct/resgroup.js with TypeScript 
- optimised some parts of resgroup.ts
- added folder and first e2e test for S-Group structures

(Screenshots, videos, or GIFs, if applicable)
Closes #2389 

## Check list
- [ ] unit-tests written
- [x] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"

Nitvex commented on packages/ketcher-core/src/domain/entities/sgroup.ts:
```diff
@@ -82,6 +82,9 @@ export class SGroup {
   pp: Vec2 | null;
   data: any;
+  dataArea: any;
+  atomId?: number;
```
What `atomId` do you mean?
What is `position`?

vitaepam replied:
You are right, I don't need this, these variables would be received as a result of getContractedPosition()
Removed

Nitvex commented on packages/ketcher-core/src/application/render/restruct/resgroup.ts:
```diff
@@ -14,25 +14,50 @@
+  atomSet: Pile;
+  bracketBox: Box2Abs;
+  d: Vec2;
+  lowerIndexText?: string | null;
```
What do you mean here by `index`?

vitaepam replied:
Index here as far as I understood from code and S-Groups behaviour is this 
<img width="504" alt="Screenshot 2023-08-14 at 20 42 34" src="https://github.com/epam/ketcher/assets/109075831/c5c5c6db-0e4a-48ff-84c1-7001c5a10cc2">

Nitvex commented on packages/ketcher-core/src/application/render/restruct/resgroup.ts:
```diff
@@ -349,55 +370,53 @@ function drawAbsoluteDat(restruct, sgroup) {
-  render,
-  id,
-) {
+  mol: any,
```
Most likely, type of mol is `Struct`

Nitvex commented on ketcher-autotests/tests/Templates/S-Groups/s-groups.spec.ts:
```diff
@@ -0,0 +1,37 @@
+    Test case related to issue: https://github.com/epam/ketcher/issues/2389
+    Description: Open file with S-groups (with Unsupported S-group type) and see that brackets in place for all S-Groups except DAT
+    */
+      await waitForLoad(page, async () => {
```
You don't need `waitForLoad` here. That's because `openFileAndAddToCanvas` already invokes that function

vitaepam replied:
Updated

vitaepam commented:
I run all tests that fail in CI locally and most of them pass (screenshots attached).
erase-tool.spec.ts - need to be reviewed and fixed, it has some strange issue. Added for now fixme()
<img width="774" alt="Screenshot 2023-08-17 at 16 05 53" src="https://github.com/epam/ketcher/assets/109075831/c804a440-dc5c-4f32-8e87-64a7f21d9e40">
<img width="622" alt="Screenshot 2023-08-17 at 16 08 42" src="https://github.com/epam/ketcher/assets/109075831/9a3f0e47-cd43-4617-9ec0-7eab3dce3a20">
<img width="590" alt="Screenshot 2023-08-17 at 16 18 27" src="https://github.com/epam/ketcher/assets/109075831/c953ecb1-9228-4e50-a159-b14e9852bfec">
<img width="593" alt="Screenshot 2023-08-17 at 16 21 12" src="https://github.com/epam/ketcher/assets/109075831/77385f0b-9c92-4f37-8dfc-2a8f2a9d09e3">
<img width="992" alt="Screenshot 2023-08-17 at 16 33 38" src="https://github.com/epam/ketcher/assets/109075831/3e9560de-c75e-4771-a99e-2155353b93b0">
<img width="536" alt="Screenshot 2023-08-17 at 16 38 11" src="https://github.com/epam/ketcher/assets/109075831/46c44993-ab30-43c0-a109-7ca8e6ccd634">
<img width="983" alt="Screenshot 2023-08-18 at 10 34 00" src="https://github.com/epam/ketcher/assets/109075831/53042a57-41a4-49ca-8d1c-a2d34c0f1b2d">
<img width="742" alt="Screenshot 2023-08-18 at 10 35 47" src="https://github.com/epam/ketcher/assets/109075831/20431a83-1cdd-4c92-aca1-4ce25aeb4835">
<img width="620" alt="Screenshot 2023-08-18 at 10 37 48" src="https://github.com/epam/ketcher/assets/109075831/1a706144-650d-457e-a897-9009310ede7c">
<img width="584" alt="Screenshot 2023-08-18 at 10 39 47" src="https://github.com/epam/ketcher/assets/109075831/82d8b1a0-f14e-4cc8-8356-af64a72b9d2b">

Zhirnoff commented:
Tested. Bug is not fixed.

https://github.com/epam/ketcher/assets/62464241/83856192-8c59-4156-a45f-f60ec0c50956

Zhirnoff commented on ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Erase-Tool/erase-tool.spec.ts:
```diff
@@ -16,6 +19,7 @@ function checkElementExists(element: RxnPlus | RxnArrow, errorMsg: string) {
 let point: { x: number; y: number };
 
 test.describe('Erase Tool', () => {
+  // TO DO: here in both tests we have some issue with openFileAndAddToCanvas() function it need proper investigation
```
What is the problem in the 'openFileAndAddToCanvas()' function?

Zhirnoff commented:
Pre-tested.
![2023-08-30_12h01_18](https://github.com/epam/ketcher/assets/62464241/70ea3b72-3858-42a6-bc45-fa4e76989f93)
